torch/testing/_internal/common_methods_invocations.py

```diff
@@ -595,19 +595,24 @@ def __init__(self,
         else:
             self.autodiff_nonfusible_nodes = autodiff_nonfusible_nodes
 
-        # autograd support
-        self.supports_autograd = supports_autograd
-        self.supports_inplace_autograd = supports_inplace_autograd
-        if self.supports_inplace_autograd is None:
-            self.supports_inplace_autograd = supports_autograd
+        # Autograd support
 
-        self.gradcheck_wrapper = gradcheck_wrapper
-        self.supports_gradgrad = supports_gradgrad
+        # Autograd flags that don't depend on backward AD
         self.supports_forward_ad = supports_forward_ad
-        self.check_batched_grad = check_batched_grad
-        self.check_batched_gradgrad = check_batched_gradgrad
-        self.gradcheck_nondet_tol = gradcheck_nondet_tol
         self.gradcheck_fast_mode = gradcheck_fast_mode
+        self.gradcheck_wrapper = gradcheck_wrapper
+
+        # Autograd flags that depend on backward AD only
+        self.supports_autograd = supports_autograd
+        self.supports_gradgrad = supports_autograd and supports_gradgrad
+        self.check_batched_grad = supports_autograd and check_batched_grad
```

Comment:
As discussed with @albanD make this default None, raise error if user specifies inconsistent value, and default to value of supports autograd otherwise